astropy/io/fits/hdu/base.py

```diff
@@ -457,7 +457,7 @@ def block_iter(nbytes):
         hdu._data_size = size + _pad_length(size)
 
         # Checksums are not checked on invalid HDU types
-        if checksum and checksum != 'remove' and isinstance(hdu, _ValidHDU):
+        if isinstance(hdu, _ValidHDU):
```

Comment:
(https://github.com/astropy/astropy/blob/master/astropy/io/fits/scripts/fitscheck.py#L143)
The script is going to access `hdu._checksum` anyways.